src/main/java/step3/domain/LottoMachine.java

```diff
@@ -15,47 +13,53 @@ public class LottoMachine {
     public static final int NUMBER_BOX_START_NUMBER = LottoNumber.NUMBER_BOX_START_NUMBER;
     public static final int NUMBER_BOX_END_NUMBER = LottoNumber.NUMBER_BOX_END_NUMBER;
 
-    private final List<Integer> numberBox;
-
-    public LottoMachine() {
-        this.numberBox = rangeClosedNumberBox();
-    }
-
-    private static List<Integer> rangeClosedNumberBox() {
+    private static List<LottoNumber> rangeClosedNumberBox() {
         return IntStream.rangeClosed(NUMBER_BOX_START_NUMBER, NUMBER_BOX_END_NUMBER)
-                .boxed()
+                .mapToObj(LottoNumber::of)
                 .collect(Collectors.toList());
     }
 
-    public Lottos play(int paidMoney){
-        validatePaidMoney(paidMoney);
-        int gameCount = paidMoney / PRICE_PER_LOTTO;
-        return lottos(gameCount);
+    public Lottos play(int paidMoney, Lottos inputManualLottos) {
+        validatePaidMoney(paidMoney, inputManualLottos);
+        return mergeLottos(autoLottosWithGameCount(getGameCount(paidMoney, inputManualLottos)), inputManualLottos);
     }
 
-    private void validatePaidMoney(int paidMoney) {
+    private void validatePaidMoney(int paidMoney, Lottos lottos) {
         if (paidMoney % PRICE_PER_LOTTO != 0 || paidMoney < PRICE_PER_LOTTO) {
             throw new IllegalArgumentException("로또 구입 금액은 " + PRICE_PER_LOTTO + "원 단위로 가능합니다. (현재 금액: " + paidMoney + "원)");
         }
     }
 
-    private Lottos lottos(int gameCount) {
-        List<Lotto> lottoList = new ArrayList<>();
+    private int getGameCount(int paidMoney, Lottos inputManualLottos) {
+        return moneyForAuto(paidMoney, inputManualLottos) / PRICE_PER_LOTTO;
+    }
+
+    private Lottos mergeLottos(Lottos autoLottos, Lottos manualLottos) {
+        List<Lotto> lottos = new ArrayList<>();
+        lottos.addAll(manualLottos.lottos());
+        lottos.addAll(autoLottos.lottos());
+        return new Lottos(lottos);
```

Comment:
```suggestion
        return Stream.concat(autoLottos.lottos().stream(), manualLottos.lottos().stream())
                .collect(Collectors.collectingAndThen(Collectors.toList(), Lottos::new));
```
빈 리스트를 만들고 추가하는 방식보다
리스트를 이어서 리스트를 만드는 게 어떨까요?